src/main/java/org/opensearch/sdk/ExtensionsRunner.java

```diff
@@ -93,6 +116,62 @@ PluginResponse handlePluginsRequest(PluginRequest pluginRequest) {
         return pluginResponse;
     }
 
+    NamedWriteableRegistryResponse handleNamedWriteableRegistryRequest(DefaultExtensionPointRequest request) {
+
+        logger.info("Registering Named Writeable Registry Request recieved from OpenSearch.");
+
+        // iterate through Extensions's named writeables and add to extension entries
+        Map<String, Class> extensionEntries = new HashMap<>();
+        for (NamedWriteableRegistry.Entry entry : this.namedWriteables) {
+            extensionEntries.put(entry.name, entry.categoryClass);
+        }
+        NamedWriteableRegistryResponse namedWriteableRegistryResponse = new NamedWriteableRegistryResponse(extensionEntries);
+        return namedWriteableRegistryResponse;
+    }
+
+    <C extends NamedWriteable> NamedWriteableRegistryParseResponse handleNamedWriteableRegistryParseRequest(
+        NamedWriteableRegistryParseRequest request
+    ) throws IOException {
+
+        logger.info("Registering Named Writeable Registry Parse request from OpenSearch");
+        boolean status = false;
+
+        // extract data from request and procress fully qualified category class name into class instance
+        Class<C> categoryClass = (Class<C>) request.getCategoryClass();
```

Comment:
For my education what's type `C`?

Reply:
Type C denotes the class of the writeable object. There are many various classes that implement the writeable class, however they are all expected to extend the NamedWriteable class. By casting the category class sent over from OpenSearch with this type, we can ensure that the category class entries extend NamedWriteables